src/Runner.Worker/StepsRunner.cs

```diff
@@ -95,7 +95,10 @@ public async Task RunAsync(IExecutionContext jobContext)
                     if (step is IActionRunner actionStep)
                     {
                         // Set GITHUB_ACTION
-                        step.ExecutionContext.SetGitHubContext("action", actionStep.Action.Name);
+                        if (!string.IsNullOrEmpty(actionStep.Action.ContextName))
```

Comment:
Lets touch base tomorrow. The rest of the PR seems fine. Not sure what to do about this. Would need scope name to disambiguate. Might be better to just leave it as refname? I dont know

Reply:
but yes contextName is null unless explicitly defined by customer so this change is better than it was

Reply:
feel free to merge since this is net improvement, we can touch base later